pcb/rules/F7_4.py

```diff
@@ -11,28 +23,85 @@ def __init__(self, module, args):
 
         self.required_layers = ["*.Cu","*.Mask"]
 
+
+
     def checkPads(self, pads):
 
         self.wrong_layers = []
 
         errors = []
 
-        for pad in pads:
-            layers = pad['layers']
+        padNoMax = 0
 
-            # For THT parts, following layers required:
-            # *.Cu
-            # F.Mask
-            # B.Mask
 
+        module = self.module
+        fpName = module.name
+
+        # Check if footprint name contains the standard text strings for
+        # footprints with exposed pad and _ThermalVias
+        # Looks for the string:
+        # '-xEP' where x is any number between 0 and 9 and designates number of exposed pads.
+        # EP is NOT case sensitive and registers 'ep' just as well as 'EP'
+        m = re.search('\\-[0-9].[a-zA-Z]', fpName)
```

Comment:
I don't understand this re. Why is there a dot?
how about 
` -\d+[EP]{2} `
 with case insensitive set?